tests/bats.lib.bash

```diff
@@ -162,6 +162,22 @@ with_kubeconfig() {
   export DETIK_CLIENT_NAME="kubectl"
 }
 
+# sets up a temporary GPG home and key for local testing
+# usage: with_temporary_gpg
+with_temporary_gpg() {
+  if ! command -v setup-local-gpg &>/dev/null; then
+    log.fatal "setup-local-gpg script not found in path"
+  fi
+
+  eval "$(setup-local-gpg)"
+
+  if [[ -z "${CK8S_PGP_FP:-}" ]]; then
+    fail "Failed to generate temporary GPG key"
+  fi
+
+  log.trace "Temporary GPG initialized. Fingerprint: $CK8S_PGP_FP"
+}
+
```

Comment:
Is this needed? We already have a function for it already in `common/bats/gpg.sh` that also takes care of an edge case in which the GPG agent isn't available in the first tries.